File: Java/Bitwise AND of Numbers Range.java
public class Solution {

    /**
     * Time:   O(log(n))
     * Memory: O(1)
     */
    public int rangeBitwiseAnd(int left, int right) {
        int i = 0;

        while (left != right) {
            left >>= 1;
            right >>= 1;
            ++i;
        }

        return left << i;
    }
}

File: Python/Bitwise AND of Numbers Range.py
class Solution:
    """
    Time:   O(log(n))
    Memory: O(1)
    """

    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        i = 0

        while left != right:
            left >>= 1
            right >>= 1
            i += 1

        return left << i
